File: IT_support _App/IT_support_App/src/main/java/com/example/IT_support/_App/entities/Ticket.java
package com.example.IT_support._App.entities;

import com.example.IT_support._App.enums.TiketStatus;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import jakarta.persistence.*;
@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Ticket {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String description;
    private LocalDate dateCreation;
    @Enumerated(EnumType.STRING)
    private TiketStatus status;
    @ManyToOne
    private User user;
    @ManyToOne
    private Technicien technicien;
    @ManyToOne
    private Panne panne;


}

File: IT-frontend/src/app/models/ticket.ts
import {Ticketstatus} from "../ennums/ticketstatus";
import {User} from "./user";
import {Technicien} from "./technicien";
import {Panne} from "./panne";

export class Ticket {
  id: number;
  title: string;
  description: string;
  dateCreation: Date;
  status: Ticketstatus;

  user: User | null;
  technicien: Technicien | null;
  panne: Panne | null;

  constructor(
    title: string,
    description: string,
    status: Ticketstatus,
    user: User,
    technicien: Technicien,
    panne: Panne,
    id: number,
    dateCreation: Date
  ) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.dateCreation = dateCreation;
    this.status = status;
    this.user = user;
    this.technicien = technicien;
    this.panne = panne;
  }
}
